File: Lab 8/t3.go
package main

import (
  "fmt"
  "os"
  "bufio"
  "strings"
)

func main() {
  os.Exit(t3_main())
}

func t3_main() int {
  var str1, str2 string
  var e error

  stdin := bufio.NewReader(os.Stdin)

  fmt.Println("Please input the first string:")
  _, e = fmt.Fscanf(stdin, "%s", &str1)
  if e != nil {
    fmt.Println("Read Error:", e)
    return 1
  }

  stdin.Discard(stdin.Buffered())

  fmt.Println("Please input the second string:")
  _, e = fmt.Fscanf(stdin, "%s", &str2)
  if e != nil {
    fmt.Println("Read Error:", e)
    return 1
  }

  // "Compare... is usually clearer and always faster to use the built-in string
  // comparison operators ==, <, >, and so on."
  // -- https://golang.org/pkg/strings/#Compare
  // if str1 == str2 {
  if strings.Compare(str1, str2) == 0 {
    fmt.Println("The two strings are the same.")
  } else {
    fmt.Println("The two strings are not the same.")
  }

  return 0
}

File: Lab 8/t3.rs
use std::io::stdin;
use std::process::exit;

fn main() {
  let mut str1 = String::new();
  println!("Please input the first string:");
  match stdin().read_line(&mut str1) {
    Ok(_) => {}
    Err(e) => {
      eprintln!("Read error: {}", e);
      exit(1);
    }
  }

  let mut str2 = String::new();
  println!("Please input the second string:");
  match stdin().read_line(&mut str2) {
    Ok(_) => {}
    Err(e) => {
      eprintln!("Read error: {}", e);
      exit(1);
    }
  }

  println!("The two strings are{} the same.",
           if str1 == str2 { "" } else { " not" });
}
